src/App.js

```diff
@@ -1,56 +1,203 @@
 import React, { Component } from 'react';
+import isEqual from 'lodash.isequal'
 
 import Banner from './Components/Header/Banner';
 import Container from './Components/DynamicContent/Container';
 import Menu from './Components/Menu/Menu';
-import {menuData} from './Components/Menu/menuData';
+import { menuData } from './Components/Menu/menuData';
 
 import './Components/Header/Banner.css';
 import './Components/Menu/Menu.css';
 import './Components/DynamicContent/Search/Search.css';
-import './Components/DynamicContent/Config/Config.css';
 import './Components/DynamicContent/Asset/AssetManager.css';
+import './Components/DynamicContent/PromotionBuilder/PromotionBuilder.css';
 import './Components/DynamicContent/Container.css';
 import './App.css';
+import { promotionData } from './Components/DynamicContent/PromotionBuilder/promotionData';
+import { getTruthyList , arrayToMap } from './utils/utils'
 
 
 class App extends Component {
 
-  state = { 
-    view: "Search"
+  state = {
+    view: "Search",
+    selectedPromotionId: null,
+    formData: {
+      id: 0,
+      devices: new Map(),
+      ventures: new Map(),
+      url: "",
+      name: "",
+      lastUpdatedTime: ""
+    },
+    searchTerm: "",
+    searchDisplay: [],
+    promotionData: promotionData
   };
-  
-  onclick = (event) => {
+
+
+  componentDidMount(){
+    this.searchPromotions();
+  }
+
+
+  componentDidUpdate(prevProps, prevState){
+    if(!isEqual(prevState.promotionData,this.state.promotionData)){
+      this.searchPromotions();
+    }
+    
+  }
+
+  handleInputChange = event => this.setState({ searchTerm: event.target.value });
```

Comment:
this can also be written as `({ target: { searchTerm } }) => this.setState({ searchTerm });`